File: JavaStudy/TopCoder/Easy/AveragePrice/AveragePrice.java
import java.util.*;

public class AveragePrice {
	public  double nonDuplicatedAverage(int[] prices) {
	   HashSet<Integer> priceCheck = new HashSet<Integer>();
	   for (int i=0; i<prices.length; i++) {
		   priceCheck.add(prices[i]);
	   }
	   double answer = 0;
	   for (int num : priceCheck) {
		   answer += num;
	   }
	   return answer/priceCheck.size();
	}
}

File: C#Study/TopCoder/Easy/AveragePrice/AveragePrice.cs
using System;
using System.Collections.Generic;

public class AveragePrice
{
    public double NonDuplicatedAverage(int[] prices)
    {
        HashSet<int> priceCheck = new HashSet<int>();
        foreach (int price in prices)
        {
            priceCheck.Add(price);
        }

        double answer = 0;
        foreach (int num in priceCheck)
        {
            answer += num;
        }

        return answer / priceCheck.Count;
    }
}
